### PR TITLE
[SYCL][E2E] increase test coverage for subregions copies

### PR DESCRIPTION
**Pull request by cppchedy**
* Updated 2D subregion copy test so that it follows the other subregion tests.
* Updated the 3D and 2D subregion tests to cover test cases for copying with non-zero offsets from device to host.

### CODE REVIEW

**Review by przemektmalon — APPROVED** [review]

**Review by Seanst98 — APPROVED** [review]

**Review by isaacault — APPROVED** [review]

**Comment by cppchedy** [comment]
The Tests failed in this [job](https://github.com/intel/llvm/actions/runs/9614095075/attempts/1#summary-26518190152) aren't related to the bindless images tests updated in this PR.

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -119,8 +119,8 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
   )
 
   fetch_adapter_source(cuda
-    ${UNIFIED_RUNTIME_REPO}
-    ${UNIFIED_RUNTIME_TAG}
+    "https://github.com/cppchedy/unified-runtime.git"
+    8e5a33b6bda832b67301277a90f641d4421ece8b
```
Comment (callumfare) [inline sycl/plugins/unified_runtime/CMakeLists.txt:123]:
```suggestion
    ${UNIFIED_RUNTIME_REPO}
    # commit 08acafb6b36c9ff12753922365df150f677655e7
    # Merge: 1e9b1b49 8e5a33b6
    # Author: Callum Fare <callum@codeplay.com>
    # Date:   Mon Jun 24 10:56:12 2024 +0100
    #     Merge pull request #1714 from cppchedy/chedy/fix-copy-in-bytes-bug
    #     [CUDA][Bindless][Exp] Fix subregion copies
    08acafb6b36c9ff12753922365df150f677655e7
```

**Comment by callumfare** [comment]
https://github.com/oneapi-src/unified-runtime/pull/1714 has merged, please pull in the latest sycl branch changes and update the UR repo/tag as suggested.

**Review by callumfare — APPROVED** [review]

**Comment by callumfare** [comment]
@intel/llvm-gatekeepers Please merge

**Pull request merged**